src/backend/gl/src/queue.rs

```diff
@@ -564,6 +564,20 @@ impl CommandQueue {
                 let gl = &self.share.context;
                 gl.BindSampler(index, sampler);
             }
+            com::Command::SetTextureSamplerSettings(index, texture, ref sinfo) => unsafe {
+                let gl = &self.share.context;
+                gl.ActiveTexture(gl::TEXTURE0 + index);
+                gl.BindTexture(gl::TEXTURE_2D, texture);
```

Comment:
we should at least warn/panic here is the texture is not 2D

Reply:
Currently the OGL backend does not keep track of dimensionality, as `create_image` currently only supports making 2D images, therefore, all textures will be 2D. All the other code in the backend also assumes textures will only be 2D. Solving this problem is beyond the scope of this PR.